### DISCUSSION TITLE
UGS with different machine profiles

### DISCUSSION THREAD

**Discussion by zutim13** [winder/Universal-G-Code-Sender#2394] [Help]
Hello, i just want to ask, if UGS is able to load different machine profiles and if i can control with UGS a fluidNC flashed ESP32?
Thanks for help!

**Comment by breiler** [comment]
Yes, UGS should work with FluidNC, make sure you select the FluidNC firmware option in the connection menu.

What do you mean by different machine profiles?

**Reply by zutim13** [reply]
Hey Joacim, thank you for the fast reply, my name is Tim!
I have an apple laptop and 3 different milling machines. all of the should get a ESP32 with fluidnc, but i just want to have one laptop and the UGS AND WANT TO LOAD DIFFERENT MACHINE PROFILES WHEN I CHANGE FROM ONE MACHINE TO THE OTHER.
I WILL ONLY USE ONE MACHINE AT TEH TIME; BUT THEY HAVE DIFFERENT STEPS PER MM AND SO ON...
Sorry, hit the capslock :)

**Reply by breiler** [reply]
OK, I understand.

No, it is not possible to have different profiles with different settings in UGS.

**Reply by zutim13** [reply]
ah ok, do you have an idea for a workaround?

**Reply by breiler** [reply]
Sorry, I may have read your post too quickly. 

If each machine has its own controller (ESP32), it will have its own configuration specific for that machine. Ie, how many motors, axes, steps per mm and if homing is enabled and so on.

**Reply by zutim13** [reply]
Ok, thank you, but i´m  a little confused, do i set things like "steps per mm" etc. in UGS or do i do it in FluidNC? And if i connect to FluidNC do i do it via USB, right?

**Reply by breiler** [reply]
Machine specific settings are done on FluidNC. You can edit the settings from UGS, but it is easier to use the online tool https://installer.fluidnc.com/.

You can connect UGS to FluidNC using both USB, Wifi or Bluetooth serial. I would recommend USB to get started.

**Reply by zutim13** [reply]
ah ok, so if i connect the UGS to the ESP32 with FluidNC on, UGS reads and show the settings done before in FluidNC? just to get it right..
Thanks!

**Reply by zutim13** [reply]
Sorry, for double asking, you answered allready.
One last question, if i use Fusion360 - UGS - FluidNC, which Post Processor do i use in Fusion360? Can i use normal "GRBL MILL" for my mill and  "GRBL LATHE" for my Lathe, or are there other PP for FluidNC on the ESP32 i should use.
Thank you!!

**Reply by breiler** [reply]
> UGS reads and show the settings done before in FluidNC?

Sorry for the late reply - yes.

Regarding the post processor, I run on Linux so I can't use Fusion360 and I am probably not the right person to answer. But it sounds resonable to use the GRBL profiles as FluidNC is based on the GRBL code base.

**Reply by zutim13** [reply]
Ok, thank you!! Have a nice christmas time!
Tim
